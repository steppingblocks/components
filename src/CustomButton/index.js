import React from 'react'
import { Button } from 'antd'
import withTheme from '../withTheme'
import { ButtonProps } from '../proptypes'

const CustomButton = withTheme(({ content, ...rest }) => (
  <Button {...rest}>{content}</Button>
))

CustomButton.propTypes = ButtonProps

CustomButton.defaultProps = {}

CustomButton.displayName = 'CustomButton'

export default CustomButton
