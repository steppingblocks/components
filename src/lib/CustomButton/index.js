import React from 'react'
import { Button } from 'antd'
import { withTheme } from '../withTheme'

const CustomButton = ({ content, ...rest }) => (
  <Button {...rest}>{content}</Button>
)

export default withTheme(CustomButton)
