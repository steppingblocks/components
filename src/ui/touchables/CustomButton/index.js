import React from 'react'
import { Button } from 'antd'
import { withTheme } from '../../../hocs/withTheme'

const CustomButton = ({ content, ...rest }) => (
  <Button {...rest}>{content}</Button>
)

export default withTheme(CustomButton)
