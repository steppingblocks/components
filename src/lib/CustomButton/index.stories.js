import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { boolean, text, select, withKnobs } from '@storybook/addon-knobs'
import CustomButton from './index'

const stories = storiesOf('CustomButton', module)

stories.add('default', () => {
  const props = {
    block: boolean('block', false),
    content: text('content', 'This is some text'),
    disabled: boolean('disabled', false),
    loading: boolean('loading', false),
    onClick: action('onClick'),
    type: select('type', [
      null,
      'primary',
      'ghost',
      'dashed',
      'danger',
      'link'
    ]),
    shape: select('shape', [null, 'circle', 'round']),
    size: select('size', [null, 'small', 'large'])
  }
  return <CustomButton {...props} />
})

stories.addDecorator(withKnobs)
