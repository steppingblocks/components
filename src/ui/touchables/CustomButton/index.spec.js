import 'jest-dom/extend-expect'
import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'
import CustomButton from '.'

afterEach(cleanup)

describe('CustomButton', () => {
  it('should render', () => {
    const content = 'Click me'
    const { container } = render(<CustomButton content={content} />)
    expect(container).toMatchSnapshot()
  })

  it('should respond to onClick events', () => {
    const content = 'Click me'
    const spy = jest.fn()
    const { getByText } = render(
      <CustomButton content={content} onClick={spy} />
    )

    fireEvent.click(getByText(content))

    expect(spy).toHaveBeenCalledTimes(1)
  })
})
