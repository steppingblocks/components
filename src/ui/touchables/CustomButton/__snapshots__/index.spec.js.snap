// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton should render 1`] = `
<div>
  <button
    class="ant-btn"
    type="button"
  >
    <span>
      Click me
    </span>
  </button>
</div>
`;
